### PR TITLE
Add pthread flag

### PR DESCRIPTION
**Pull request by seelabs**
@ximinez @bachase

### CODE REVIEW

**Comment by bachase** [comment]
👍 tested cmake debug unity in Ubuntu VM.

File: Builds/CMake/CMakeFuncs.cmake [hunk 1]
```diff
@@ -457,6 +457,9 @@ macro(setup_build_boilerplate)
       add_definitions(-DSANITIZER=TSAN)
     endif()
   endif()
+  if (NOT WIN32)
+    add_compile_options("-pthread")
```
Comment (mellery451) [inline Builds/CMake/CMakeFuncs.cmake]: It's not obvious from the FindThreads docs, but I think you can alternatively use:     

`add_compile_options(${CMAKE_THREAD_LIBS_INIT})`

Also, would it make more sense to add this to `find_pthread` macro on line 293?

**Comment by nbougalis** [comment]
:+1: from me.

File: Builds/CMake/CMakeFuncs.cmake (hunk 1 above)
Comment (seelabs) [inline Builds/CMake/CMakeFuncs.cmake]: @mellery451 Good suggestions. Done.

Bike shed: We may want to rename the `find_*` functions to `use_*`. Rational: A function that "finds" something should be side-effect free (other than setting results variables for the various directories and flags), it shouldn't change any variables that  effect the target. That's the reason I didn't put it in `find_pthread` originally. Looking at `find_boost` however shows we're already doing this. I don't want to do the rename for this PR, but maybe in a later re-factor.

**Comment by seelabs** [comment]
Squashed

**Comment by nbougalis** [comment]
Merged as 810a6b0f30db5d90c5c8996189b9166516af1b8f

**Pull request closed without merging**